### PR TITLE
Bluetooth: Controller: Fix per adv chain broken after per adv re-enable

### PR DESCRIPTION
**Pull request by ppryga-nordic**
If there is enabled support for ADI in periodic advertising and
periodic advertising uses chained PDUs there is an assert when
periodic advertising is re-enabled.

There is missing a code that will set a chain for PDU with
updated ADI field.

The commit fixes the issue.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

### CODE REVIEW

**Review by kruithofa — DISMISSED** [review]

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 1]
```diff
@@ -1818,3 +1815,26 @@ static void ticker_op_cb(uint32_t status, void *param)
...
 }
+
+/**
+ * @brief Set a periodic advertising chained PDUs in a new periodic advertising data if there are
+ *        chained PDUs in former periodic advertising data.
+ *
+ * @param pdu_new  Pointer to new pdu_adv where to add chained pdu_adv.
+ * @param pdu_prev Pointer to former pdu_adv where to add chained pdu_adv.
+ */
+void per_adv_data_set_chain_from_prev_pdu(struct pdu_adv *pdu_new, struct pdu_adv *pdu_prev)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:1826]:
```suggestion
static void adv_sync_pdu_chain_check_and_duplicate(struct pdu_adv *pdu_new, struct pdu_adv *pdu_prev)
```

**Review by cvinayak — CHANGES_REQUESTED** [review]
Minor change, otherwise LGTM.

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c (hunk 1 above)
Comment (ppryga) [reply subsys/bluetooth/controller/ll_sw/ull_adv_sync.c]: Corrected

File: subsys/bluetooth/controller/ll_sw/ull_adv_sync.c [hunk 2]
```diff
@@ -1818,3 +1815,26 @@ static void ticker_op_cb(uint32_t status, void *param)
...
 }
+
+/**
+ * @brief Set a periodic advertising chained PDUs in a new periodic advertising data if there are
+ *        chained PDUs in former periodic advertising data.
+ *
+ * @param pdu_new  Pointer to new pdu_adv where to add chained pdu_adv.
+ * @param pdu_prev Pointer to former pdu_adv where to add chained pdu_adv.
+ */
+static void per_adv_data_set_chain_from_prev_pdu(struct pdu_adv *pdu_new, struct pdu_adv *pdu_prev)
```
Comment (cvinayak) [inline subsys/bluetooth/controller/ll_sw/ull_adv_sync.c:1826]: Do not use `per` as it is acronym for peripheral role in Bluetooth Test Spec, use `adv_sync` namespace as in other functions. `adv_sync_pdu_chain_check_and_duplicate` conveys the purpose of the function as it uses `adv_sync_pdu_duplicate_chain`

```suggestion
static void adv_sync_pdu_chain_check_and_duplicate(struct pdu_adv *pdu_new, struct pdu_adv *pdu_prev)
```

Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ull_adv_sync.c]: The original comment in code go this way:
https://github.com/zephyrproject-rtos/zephyr/pull/43443/files#diff-3184590f25e7d9697815d4c8e4bc868dd3429067a409add63215421214f60eecL485-L489

new `PDU` having `chain` is `check`-ed, then `no need to deal with chain as it's already there`

**Review by cvinayak — APPROVED** [review]

**Comment by ppryga** [comment]
@kruithofa could you take a look on the PR again?

**Review by kruithofa — APPROVED** [review]

**Pull request merged**